### ISSUE TITLE
Create integration test cases for various memory metrics sources

### ISSUE THREAD

**Issue by omokpo** [perfpod/memory-collector#5]
> > Let's set up GitHub actions and run the build/test/deploy cycle, assuming the Influxdb collector is part of the CI. This will enable us to flesh out the integration test framework, maybe initially with integration tests around the Influxdb collector and then adding support to the test framework for the memory collector.
> 
> Agreed! Seeing what Telegraf has should give us a good understanding of what's missing and how to gather that.

We should also create a separate issue to document specific integration test cases that should run once the cluster is up with collectors, starting with the telegraf collector. Two separate integration cases could be for the different CPU vendors (Intel RDT and AMD QoS ). Then, up the stack...

_Originally posted by @amosomokpo in https://github.com/perfpod/memory-collector/issues/2#issuecomment-2460342872_

**Comment by yonch** [comment]
I think this is currently blocked by #2 and #4

**Comment by yonch** [comment]
this might be too early -- we do not yet have any working collectors, nor noise generators that generate a signal we can measure.. closing for now (if I got this wrong, please reopen)
